watertap/examples/flowsheets/electrodialysis/electrodialysis_1stack.py

```diff
@@ -0,0 +1,326 @@
+from idaes.core.util.model_statistics import degrees_of_freedom, report_statistics
+from idaes.models.unit_models import Feed, Product, Separator
+from pandas import DataFrame
+import idaes.core.util.scaling as iscale
+import idaes.logger as idaeslogger
+from watertap.unit_models.electrodialysis_1D import Electrodialysis1D
+from watertap.costing.watertap_costing_package import (
+    WaterTAPCosting,
+)
+from watertap.property_models.ion_DSPMDE_prop_pack import DSPMDEParameterBlock
+
+__author__ = "Xiangyu Bi"
+
+
+def main():
+    # set up solver
+    solver = get_solver()
+    # Simulate a fully defined operation
+    m = build()
+    set_operating_conditions(m)
+    initialize_system(m, solver=solver)
+    solve(m, solver=solver)
+
+    print("\n***---Simulation results---***")
+    display_model_metrics(m)
+
+    # Perform an optimization over selected variables
+    initialize_system(m, solver=solver)
+    optimize_system(m, solver=solver)
+    print("\n***---Optimization results---***")
+    display_model_metrics(m)
+
+
+def build():
+    # ---building model---
+    m = ConcreteModel()
+    m.fs = FlowsheetBlock(default={"dynamic": False})
+    ion_dict = {
+        "solute_list": ["Na_+", "Cl_-"],
+        "mw_data": {"H2O": 18e-3, "Na_+": 23e-3, "Cl_-": 35.5e-3},
+        "electrical_mobility_data": {"Na_+": 5.19e-8, "Cl_-": 7.92e-8},
+        "charge": {"Na_+": 1, "Cl_-": -1},
+    }
+    m.fs.properties = DSPMDEParameterBlock(default=ion_dict)
+    m.fs.costing = WaterTAPCosting()
+    m.fs.feed = Feed(default={"property_package": m.fs.properties})
+    m.fs.separator = Separator(
+        default={
+            "property_package": m.fs.properties,
+            "outlet_list": ["inlet_diluate", "inlet_concentrate"],
+        }
+    )  # "inlet_diluate" and "inlet_concentrate" are two separator's outlet ports that are connected to the two inlects of the ED stack.
+
+    # Add electrodialysis (ED) stacks
+    m.fs.EDstack = Electrodialysis1D(
+        default={
+            "property_package": m.fs.properties,
+            "operation_mode": "Constant_Voltage",
+            "finite_elements": 20,
+        },
+    )
+    m.fs.product = Product(default={"property_package": m.fs.properties})
+    m.fs.disposal = Product(default={"property_package": m.fs.properties})
+
+    # Build extra properties as needed
+    m.fs.feed.properties[0].conc_mass_phase_comp[...]
+    m.fs.separator.inlet_diluate_state[0].conc_mass_phase_comp[...]
+    m.fs.separator.inlet_concentrate_state[0].conc_mass_phase_comp[...]
+    m.fs.product.properties[0].conc_mass_phase_comp[...]
+    m.fs.disposal.properties[0].conc_mass_phase_comp[...]
+
+    m.fs.feed.properties[0].flow_vol_phase[...]
+    m.fs.separator.inlet_diluate_state[0].flow_vol_phase[...]
+    m.fs.separator.inlet_concentrate_state[0].flow_vol_phase[...]
+    m.fs.product.properties[0].flow_vol_phase[...]
+    m.fs.disposal.properties[0].flow_vol_phase[...]
```

Comment:
Why are you "building" these? Is it for initialization or displaying?

I understand why you them to be created before scaling. I usually have a comment that says "touching variables for initialization and displaying results" rather than "build extra properties as needed"